src/harness/sas_testcase.py

```diff
@@ -141,3 +141,24 @@ def TriggerDailyActivitiesImmediatelyAndWaitUntilComplete(self):
     while not self._sas_admin.GetDailyActivitiesStatus()['completed']:
       time.sleep(10)
     signal.alarm(0)
+    
+  def TriggerFullActivityDumpAndWaitUntilComplete(self):
+    self._sas_admin.TriggerFullActivityDump()
+    request_time = datetime.utcnow()
+    
+    signal.signal(signal.SIGALRM,
+                  lambda signum, frame:
+                  (_ for _ in ()).throw(Exception('Full Activity Dump Check Timeout')))
+    dump_message = self._sas.GetFullActivityDump()
```

Comment:
Prefer to move this and the following line to just before the 'while' loop so that it's clearer you really wanted a 'do while' loop (silly Python). This will also help the two from diverging if we ever need to update the code.

Reply:
I updated the code, trying to do something like (do while) :) , I hope it is better now.

Reply:
It looks much nicer -- thanks!